africanus/calibration/utils/utils.py

```diff
@@ -39,15 +39,23 @@ def check_type(jones, vis, vis_type='vis'):
     return mode
 
 
-def chunkify_rows(time, utimes_per_chunk):
+def chunkify_rows(time, utimes_per_chunk, daskify_idx=False):
     utimes, time_bin_counts = np.unique(time, return_counts=True)
     n_time = len(utimes)
+    if utimes_per_chunk == 0 or utimes_per_chunk == -1:
+        utimes_per_chunk = n_time
     row_chunks = [np.sum(time_bin_counts[i:i+utimes_per_chunk])
                   for i in range(0, n_time, utimes_per_chunk)]
     time_bin_indices = np.zeros(n_time, dtype=np.int32)
     time_bin_indices[1::] = np.cumsum(time_bin_counts)[0:-1]
     time_bin_indices = time_bin_indices.astype(np.int32)
     time_bin_counts = time_bin_counts.astype(np.int32)
+    if daskify_idx:
+        import dask.array as da
+        time_bin_indices = da.from_array(
+            time_bin_indices, chunks=utimes_per_chunk)
+        time_bin_counts = da.from_array(
+            time_bin_counts, chunks=utimes_per_chunk)
```

Comment:
Hehe, I'm so on the fence about this. On the one hand it's probably very convenient, on the other hand it feels like this dask wrapping functionality should be elsewhere. A `dask_chunkify_rows` wrapper around `chunkify_rows` might be "better"? But I don't feel strongly about this, its a utility function after all so happy to go with whatever you decide here.

Reply:
I have removed it, was just getting repetitive but I agree it's better to do it externally